src/controller.cpp

```diff
@@ -75,7 +82,29 @@ void Controller::setPlan(const nav_msgs::msg::Path & path) { path_handler_.setPa
 
 void Controller::setSpeedLimit(const double & speed_limit, const bool & percentage)
```

Comment:
Its an API from the Nav2 level that needs to be implemented for a `Controller` type plugin in galactic and newer. It makes it so that the Speed Restricted Layer (or other external servers) can dynamically adjust the speed limit of the robot through a common interface. The controller server receives a request to change the speed limit and then it propagates that request through all of the controller plugins, where this is one of many. 

This is just how the API was designed, which could be subject to change. The implementation is also just mirroring the implementation in other controller plugin types to have consistent behavior.

Once we add dynamic parameters, it will be possible to change just X or Y or angular speed, but this callback is for scaling the entire robot behavior for safety zones